src/test/scala/pokerData/BoardGenerators.scala

```diff
@@ -2,43 +2,43 @@ package pokerData
 
 import cats.effect.unsafe.implicits.global
 import org.scalacheck.Gen.choose
-import org.scalacheck.Gen
+import org.scalacheck._
 import poker.Deck.PreflopDeck
 import poker.Street
-import poker.Street.{deal, dealHoleCards, Flop, Preflop, River, Turn}
-//import poker.Deck.makeStartingDeck
+import poker.Street._
 import poker.Deck.PreflopDeck._
 
 object BoardGenerators {
 
   val genNumberOfPlayers: Gen[Int] =
     choose(2, 10)
-  val genPreflopBoard: Gen[Preflop] = {
 
-    for {
-      numPlayers <- genNumberOfPlayers
-    } yield dealHoleCards(numPlayers).unsafeRunSync()
+  def genPreflopBoard(numPlayers: Int): Gen[Preflop] =
+    dealHoleCards(numPlayers).unsafeRunSync()
```

Comment:
A fun way to tighten this up would be to make sure that `dealHoleCards` is written in terms of `[F[_]: Random]` instead of hardcoded `IO`, and then implement `Random[Gen]`. That way you could run `dealHoleCards` either as typical imperative randomness, or as a way of generating arbitraries. I think that's a very cool possibility.